test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs

```diff
@@ -0,0 +1,866 @@
+            _jsonCompilationOptions.Add("nowarn", new JArray(_noWarnings));
+            _jsonCompilationOptions.Add("additionalArguments", new JArray(_someAdditionalArguments));
+            _jsonCompilationOptions.Add("languageVersion", SomeLanguageVersion);
+            _jsonCompilationOptions.Add("outputName", SomeOutputName);
+            _jsonCompilationOptions.Add("platform", SomePlatform);
+            _jsonCompilationOptions.Add("keyFile", SomeKeyFile);
+            _jsonCompilationOptions.Add("debugType", SomeDebugType);
+            _jsonCompilationOptions.Add("allowUnsafe", true);
+            _jsonCompilationOptions.Add("warningsAsErrors", true);
+            _jsonCompilationOptions.Add("optimize", true);
+            _jsonCompilationOptions.Add("delaySign", true);
+            _jsonCompilationOptions.Add("publicSign", true);
+            _jsonCompilationOptions.Add("emitEntryPoint", true);
+            _jsonCompilationOptions.Add("xmlDoc", true);
+            _jsonCompilationOptions.Add("preserveCompilationContext", true);
+
+            commonCompilerOptions = new CommonCompilerOptions
+            {
+                Defines = _someDefines,
+                SuppressWarnings = _noWarnings,
+                AdditionalArguments = _someAdditionalArguments,
+                LanguageVersion = SomeLanguageVersion,
+                OutputName = SomeOutputName,
+                Platform = SomePlatform,
+                KeyFile = SomeKeyFile,
+                DebugType = SomeDebugType,
+                AllowUnsafe = true,
+                WarningsAsErrors = true,
+                Optimize = true,
+                DelaySign = true,
+                PublicSign = true,
+                EmitEntryPoint = true,
+                GenerateXmlDocumentation = true,
+                PreserveCompilationContext = true
+            };
+        }
+
+        [Fact]
+        public void It_does_not_throw_when_the_project_json_is_empty()
+        {
+            var json = new JObject();
+            Action action = () => GetProject(json);
+
+            action.ShouldNotThrow<Exception>();
+        }
+
+        [Fact]
+        public void It_sets_Name_to_the_passed_ProjectName_if_one_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.Name.Should().Be(ProjectName);
+        }
+
+        [Fact]
+        public void It_sets_Name_to_the_Name_in_the_ProjectJson_when_one_is_set()
+        {
+            const string nameInprojectJson = "some name in the project.json";
+            var json = new JObject();
+            json.Add("name", nameInprojectJson);
+            var project = GetProject(json);
+
+            project.Name.Should().Be(nameInprojectJson);
+        }
+
+        [Fact]
+        public void It_sets_the_project_file_path()
+        {
+            _emptyProject.ProjectFilePath.Should().Be(ProjectFilePath);
+        }
+
+        [Fact]
+        public void It_sets_the_version_to_one_when_it_is_not_set()
```

Comment:
This is a huge test class. Should we make two, one for defaults and one for overrides?